### PR TITLE
[numpy] add logical op

### PR DESCRIPTION
**Pull request by Yiyan66**
## Description ##
add op logical_and, logical_or, logical_xor

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at https://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by hgt312** [comment]
Are the dispatch tests migrated from official numpy?

File: contrib/tvmop/core/umath.py [hunk 1]
```diff
@@ -25,6 +25,9 @@
     'less': lambda a, b, *idx: a[idx] < b[idx],
     'greater_equal': lambda a, b, *idx: a[idx] >= b[idx],
     'less_equal': lambda a, b, *idx: a[idx] <= b[idx],
+    'logical_and': lambda a, b, *idx: tvm.all(a[idx]!=0,b[idx]!=0),
+    'logical_or': lambda a, b, *idx: tvm.any(a[idx]!=0,b[idx]!=0),
+    'logical_xor': lambda a, b, *idx: tvm.all(tvm.any(a[idx]!=0,b[idx]!=0),tvm.any(a[idx]==0,b[idx]==0)),
```
Comment (hgt312) [inline contrib/tvmop/core/umath.py]: Add a white space after a comma.

File: contrib/tvmop/core/umath.py [hunk 2]
```diff
@@ -87,6 +90,9 @@ def _binary_logic_gpu(compute_func, op, itype, ndim):
     'less_scalar': lambda a, b, *idx: a[idx].astype(b.dtype) < b,
     'greater_equal_scalar': lambda a, b, *idx: a[idx].astype(b.dtype) >= b,
     'less_equal_scalar': lambda a, b, *idx: a[idx].astype(b.dtype) <= b,
+    'logical_and_scalar': lambda a, b, *idx: tvm.all(a[idx].astype(b.dtype)!=0,b!=0),
```
Comment (hgt312) [inline contrib/tvmop/core/umath.py]: Same here.

File: tests/python/unittest/test_numpy_interoperability.py [hunk 3]
```diff
@@ -1455,6 +1458,26 @@ def _add_workload_less_equal(array_pool):
     # OpArgMngr.add_workload('less_equal', np.array([np.nan]), np.array([np.nan]))
 
 
+def _add_workload_logical_and(array_pool):
+    #OpArgMngr.add_workload('logical_and', np.array([0, 1, 2, 4, 2], dtype=np.float16), np.array([-2, 5, 1, 4, 3], dtype=np.float16))
```
Comment (haojin2) [inline tests/python/unittest/test_numpy_interoperability.py]: remove unused code.

File: tests/python/unittest/test_numpy_op.py [hunk 4]
```diff
@@ -5263,48 +5267,6 @@ def dbg(name, data):
                     assert_almost_equal(grad[0][iop], grad[1][iop], rtol=rtol, atol=atol)
 
 
-@with_seed()
```
Comment (haojin2) [inline tests/python/unittest/test_numpy_op.py]: Why removing this test? Please add it back

**Pull request closed without merging**